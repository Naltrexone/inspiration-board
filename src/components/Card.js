import React from 'react';
import PropTypes from 'prop-types';

import './Card.css';

const Card = (props) => {

  console.log(typeof props.deleteCard);

  const {text, emoji } =  props.card;

  return(
    <div className="card">

      <div className="card__content">
        <div className="card__content-emoji">
          {emoji}
        </div>
        <div className="card__content-text">
          {text}
        </div>
        <button
          className="card__delete"
          onClick={props.deleteCard}>
          Delete
        </button>
      </div>
    </div>
  );
};

Card.propTypes = {
  card: PropTypes.object.isRequired,
  deleteCard: PropTypes.func.isRequired
};

export default Card;
