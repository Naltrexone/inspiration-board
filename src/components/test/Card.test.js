import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {

    const testCard = {
      id: 99,
      text: 'be happy!'
    }
    const wrapper = shallow(
      <Card card={testCard} deleteCard={ () => {} }/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
