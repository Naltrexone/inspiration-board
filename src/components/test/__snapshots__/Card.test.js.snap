// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
        Object {
            "id": 99,
            "text": "be happy!",
          }
    }
    deleteCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card__content"
>
        <div
                className="card__content-emoji"
        />
        <div
                className="card__content-text"
        >
                be happy!
        </div>
        <button
                className="card__delete"
                onClick={[Function]}
        >
                Delete
        </button>
</div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card__content-emoji"
/>,
          <div
            className="card__content-text"
>
            be happy!
</div>,
          <button
            className="card__delete"
            onClick={[Function]}
>
            Delete
</button>,
        ],
        "className": "card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "be happy!",
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": "be happy!",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "className": "card__delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card__content"
>
          <div
                    className="card__content-emoji"
          />
          <div
                    className="card__content-text"
          >
                    be happy!
          </div>
          <button
                    className="card__delete"
                    onClick={[Function]}
          >
                    Delete
          </button>
</div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card__content-emoji"
/>,
            <div
              className="card__content-text"
>
              be happy!
</div>,
            <button
              className="card__delete"
              onClick={[Function]}
>
              Delete
</button>,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "be happy!",
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": "be happy!",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete",
              "className": "card__delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
